source/app/service-providers/fsal/fsal-directory.ts

```diff
@@ -181,7 +190,21 @@ async function parseSettings (dir: DirDescriptor): Promise<void> {
  *
  * @return  {Promise<DirDescriptor>}           Resolves with the directory descriptor
  */
-export async function parse (currentPath: string, cache: FSALCache, parent: DirDescriptor|null): Promise<DirDescriptor> {
+export async function parse (
```

Comment:
That's pretty heavy. Can some of these properties be reused (say in a `context` argument) among the methods? But even for a single method something like `parse("path", context: { currentPath, cache, ... })` might be more readable.

Reply:
That's what I'm currently doing. I figured out already a lot of different ways to remove this dependency propagation deep into the FSAL. I'm not yet done doing this, but I'm getting there. It turns out, removing the side effects on the globals opens up a lot of avenues for better code.

I want to avoid `context` objects for as long as possible, because that hides a lot of the dependencies again.

Reply:
> Then you only need to pass the dependencies once in the constructor, and not for every function call.

I thought about that, but I figured that, by using closures instead, we will end up having a little more wiggle room for potential customizations, since we can always pass those functions which we actually want instead of calling a function on these objects. But I mean we can definitely look at how it looks after we're done.

What would be cool is if you could have a look at the appServiceContainer if that's feasible as I did that right now, or if I should do something else?